x-pack/platform/plugins/shared/fleet/public/applications/integrations/sections/epm/screens/installed_integrations/hooks/use_installed_integrations.tsx

```diff
@@ -0,0 +1,135 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useMemo } from 'react';
+import semverLt from 'semver/functions/lt';
+import { Search as LocalSearch, PrefixIndexStrategy } from 'js-search';
+
+import type { PackageListItem } from '../../../../../../../../common';
+import { useGetPackagesQuery, type Pagination } from '../../../../../../../hooks';
+import type {
+  InstalledIntegrationsFilter,
+  PackageInstallationStatus,
+  PackageListItemWithExtra,
+} from '../types';
+
+function getIntegrationStatus(item: PackageListItem): PackageInstallationStatus {
+  if (item.status === 'install_failed') {
+    return 'install_failed';
+  } else if (item.status === 'installed') {
+    const isUpgradeFailed = item?.installationInfo?.latest_install_failed_attempts?.some(
+      (attempt) =>
+        item.installationInfo && semverLt(item.installationInfo.version, attempt.target_version)
+    );
+
+    const isUpgradeAvailable =
+      (item?.installationInfo && semverLt(item.installationInfo.version, item.version)) ?? false;
+
+    return isUpgradeFailed
+      ? 'upgrade_failed'
+      : isUpgradeAvailable
+      ? 'upgrade_available'
+      : 'installed';
+  }
+
+  return item.status ?? 'not_installed';
+}
+
+const fieldsToSearch = ['name', 'title', 'description'];
+function useLocalSearch(packageList: PackageListItemWithExtra[], isInitialLoading: boolean) {
```

Comment:
is it possible to use the existing `useLocalSearch` hook? (at `x-pack/platform/plugins/shared/fleet/public/applications/integrations/hooks/use_local_search.tsx`)

Reply:
Done in https://github.com/elastic/kibana/pull/212582/commits/81a06e7913f74042c3b5ad88fb0a7bae22af3e52